osutil/mapper.go

```diff
@@ -0,0 +1,57 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"fmt"
+	"os/exec"
+)
+
+// CreateLinearMapper creates a linear device mapping of the given device
+// with the given offset and size.
+//
+// The mapper device node is returned.
+func CreateLinearMapper(device, name, uuid string, offset, size uint64) (string, error) {
+	errPrefix := "cannot create mapper: "
+
+	if offset%512 != 0 {
+		return "", fmt.Errorf(errPrefix+"offset %v must be aligned to 512 bytes", offset)
+	}
+	if size%512 != 0 {
+		return "", fmt.Errorf(errPrefix+"size %v must be aligned to 512 bytes", size)
+	}
+	if size <= offset {
+		return "", fmt.Errorf(errPrefix+"size %v must be larger than the offset %v", size, offset)
+	}
+
+	offsetInBlocks := offset / uint64(512)
+	sizeWithoutOffsetInBlocks := (size / uint64(512)) - offsetInBlocks
```

Comment:
golang question: are these casts really needed, given that the first operand is already a uint64?

(I'm wondering because you didn't use a cast before, when doing the modulo operation)

Reply:
Also, I'm a bit confused by the subtraction here. When a function takes two parameters called "offset" and "size", I always assume that we are talking about `size` bytes starting at offset `offset`, whereas this function uses `size - offset` bytes starting at `offset`. I suggest removing the subtraction from here.

Reply:
I was wondering if I should rename it to "totalSize" for clarity? The usage of this is usually a fixed offset and the size of the disk. But if everyone else feels it's clearer if the size is the size without the offset that should be passed in I'm happy to change the code.

Reply:
+1 for totalSize, I was confused as well.